test/fail_compilation/test23017.d

```diff
@@ -0,0 +1,20 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/test23017.d(16): Error: class `test23017.CppChildA` with C++ linkage cannot inherit from class `DClass` with D linkage
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=23017
+// C++ class may not derive from D class
+extern(D) class DClass {}
+extern(C++) class CppClass
+{
+	void foo();
+}
+
+extern(C++) class CppChildA : DClass {}
+extern(C++) class CppChildB : CppClass {}
+
+extern(D) class DChildA : DClass {} // error
+extern(D) class DChildB : CppClass {} // automatically made extern(C++)
```

Comment:
what effect does extern(D) have here? mangling? or is this just for convenience?

Reply:
Just being explicit. Apparently, `extern(D)` and nothing/default is treated identically, which surprised me: even when explicitly marked `extern(D)`, it can actually result in `extern(C++)` class kind and mangling, but it's how it is.